resources/opt/sonatype/nexus/resources/nexusSetupCompactBlobstoreTask.groovy

```diff
@@ -0,0 +1,30 @@
+import org.sonatype.nexus.scheduling.TaskScheduler
+import org.sonatype.nexus.scheduling.TaskConfiguration
+import groovy.json.JsonSlurper
+
+
+// get parameters from payload JSON file
+def configurationParameters = new JsonSlurper().parseText(args)
+
+def createCompactBlobstoreTask(configurationParameters) {
+    def taskScheduler = container.lookup(TaskScheduler.class.getName())
+
+    def allreadyExistingTask = taskScheduler.getTaskByTypeId(configurationParameters.type)
+    if (allreadyExistingTask) {
+        // already defined this kind of task
+        return
+    }
+
+    TaskConfiguration config = taskScheduler.createTaskConfigurationInstance(configurationParameters.type)
+    config.setEnabled(configurationParameters.enabled.toBoolean())
+    config.setName(configurationParameters.name)
+    config.setString("blobstoreName", configurationParameters.blobstore)
+
+    Calendar today = Calendar.getInstance();
```

Comment:
Usually I'd suggest Javas new Time API here, but from the looks it would be awkward to convert it back to `java.util.date`...